src/Sarif.Driver/Sdk/ExtensionMethods.cs

```diff
@@ -1,13 +1,118 @@
+            //    (startLine,startColumn)
+            //    (startLine,startColumn-endColumn)
+            //    (startLine,startColumn,endLine,endColumn)
+
+            bool multiline = region.EndLine > region.StartLine;
+            bool multicolumn = (multiline || region.EndColumn > region.StartColumn);
+
+            if (multiline)
+            {
+                if (multicolumn && (region.StartColumn > 1 || region.EndColumn > 1))
+                {
+                    //  (startLine,startColumn,endLine,endColumn)
+                    return
+                        "(" +
+                        region.StartLine.ToString() + "," +
+                        (region.StartColumn > 0 ? region.StartColumn.ToString() : "1") + "," +
+                        region.EndLine.ToString() + "," +
+                        (region.EndColumn > 0 ? region.EndColumn.ToString() : "1") +
+                        ")";
+                }
+                //  (startLine-endLine)
+                return
+                    "(" +
+                    region.StartLine.ToString() + "-" + region.EndLine.ToString() +
+                    ")";
+            }
+
+            if (multicolumn)
+            {
+                // (startLine,startColumn-endColumn)
+                return
+                    "(" +
+                    region.StartLine.ToString() + "," +
+                    region.StartColumn.ToString() + "-" +
+                    region.EndColumn.ToString() +
+                    ")";
+            }
+
+            if (region.StartColumn > 1)
+            {
+                // (startLine,startColumn)
+                return
+                     "(" +
+                     region.StartLine.ToString() + "," + region.StartColumn.ToString() +
+                     ")";
+            }
+            // (startLine)
+            return
+                 "(" +
+                 region.StartLine.ToString() + "," + region.StartColumn.ToString() +
+                 ")";
+        }
+
+        public static string FormatForVisualStudio(this Result result, IRuleDescriptor rule)
+        {
+            var messageLines = new List<string>();
+            foreach (var location in result.Locations)
+            {
+                var components = location.ResultFile ?? location.AnalysisTarget;
+                var lastComponent = components.Last();
+                messageLines.Add(
+                    string.Format(
+                        CultureInfo.InvariantCulture, "{0}:{1} {2} {3}: {4}",
+                        lastComponent.Uri.AbsolutePath,
+                        lastComponent.Region.FormatForVisualStudio(),
+                        result.Kind.FormatForVisualStudio(),
+                        result.RuleId,
+                        result.GetMessageText(rule)
+                        ));
+            }
+
+            return string.Join("\n", messageLines);
+        }
+
+        public static string FormatForVisualStudio(this ResultKind kind)
+        {
```

Comment:
I don't _think_ these should be localized because people use tools to parse the log file output. What do you think?